hpedockerplugin/hpe/hpe_3par_common.py

```diff
@@ -938,3 +967,199 @@ def _get_prioritized_host_on_3par(self, host, hosts, hostname):
+                             "version '%(version)s' is installed.") %
+                           {'dedup_version': DEDUP_API_VERSION,
+                            'version': self.API_VERSION})
+                    LOG.error(err)
+                    raise exception.InvalidInput(reason=err)
+
+                compression_val = src_vref['compression']  # None/true/False
+                compression = None
+                if compression_val is not None:
+                    compression = self.get_compression_policy(compression_val)
+
+                # make the 3PAR copy the contents.
+                # can't delete the original until the copy is done.
+                self._copy_volume(src_3par_vol_name, dst_3par_vol_name,
+                                  cpg=cpg, snap_cpg=snap_cpg,
+                                  tpvv=tpvv, tdvv=tdvv,
+                                  compression=compression)
+
+                # Check if flash cache needs to be enabled
+                flash_cache = \
+                    self.get_flash_cache_policy(src_vref['flash_cache'])
+
+                if flash_cache is not None:
+                    try:
+                        self._add_volume_to_volume_set(dst_volume,
+                                                       src_3par_vol_name,
+                                                       cpg, flash_cache)
+                    except exception.InvalidInput as ex:
+                        # Delete volume if unable to add it to volume set
+                        self.client.deleteVolume(dst_3par_vol_name)
+                        LOG.error(_LE("Exception: %s"), ex)
+                        raise exception.PluginException(ex)
+            else:
+                # The size of the new volume is different, so we have to
+                # copy the volume and wait.  Do the resize after the copy
+                # is complete.
+                LOG.debug("Creating a clone of volume, using offline copy.")
+
+                # we first have to create the destination volume
+                model_update = self.create_volume(dst_volume)
+
+                optional = {'priority': 1}
+                body = self.client.copyVolume(src_3par_vol_name,
+                                              dst_3par_vol_name, None,
+                                              optional=optional)
+                task_id = body['taskid']
+
+                task_status = self._wait_for_task_completion(task_id)
+                if task_status['status'] is not self.client.TASK_DONE:
+                    dbg = {'status': task_status, 'id': dst_volume['id']}
+                    msg = _('Copy volume task failed: create_cloned_volume '
+                            'id=%(id)s, status=%(status)s.') % dbg
+                    raise exception.PluginException(msg)
+                else:
+                    LOG.debug('Copy volume completed: create_cloned_volume: '
```

Comment:
you can make to LOG.info() since for code tracing it will help if the logging is restrictive.